Sources/NIO/ChannelOption.swift

```diff
@@ -186,6 +186,36 @@ public enum AllowRemoteHalfClosureOption: ChannelOption {
     case const(())
 }
 
+/// `UserDefinedWritabilityOption` allows users to control the `isWritable` status of the channel aside from its
+/// normal calculation.  This is useful for handlers that need to have a say in when the channel is writable. For
+/// example, the traffic shaping handler needs to be able to disable writability at certain times and this option
+/// provides that capability.
+///
+/// There are 63 available flags and users must negotatiate an index on their own. The option carries the index
```

Comment:
I don't really love having users negotiate an index on their own. I'd vastly prefer users being able to ask for an index. I appreciate that that will require a bigger change, and we don't necessarily need to do it right away, but I don't love that at all.

Reply:
I'm in favour of asking rather than the user negotiating as well, think we should do this straight away.